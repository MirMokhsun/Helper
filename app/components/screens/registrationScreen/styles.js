import { StyleSheet } from 'react-native';
import Utils from '../../../api/utils/utils';

export const styles = StyleSheet.create({
    scrollview: {
        flex: 1,
        height: "100%",
        width: "100%",
    },
    wrapperAbsolute: {
        position: "absolute",
        width: "100%",
        height: "100%",
        top: 0,
        left: 0,
        backgroundColor: '#E5E5E5',
    },
    container: {
        width: "90%",
        minHeight: Utils.size.height * 0.7, // 70% need for IphoneX
        flexDirection: 'column',
        alignItems: 'center',
        backgroundColor: '#FFFFFF',
        borderRadius: 10,
        justifyContent: 'space-around',
    },
    registrationTitle: {
        fontFamily: 'Roboto',
        fontWeight: 'bold',
        lineHeight: 32,
        fontSize: 22,
        textAlign: 'center',
        color: '#000756',
        marginVertical: 25,
    },
    backgroundBlue: {
        position: 'absolute',
        top: -50,
        left: 0
    },
    logoWrapper: {
        alignItems: 'center',
        justifyContent: 'center',
        height: 35,
        width: '100%',
        marginBottom: 20,
        marginTop: Utils.isIphoneX ? 60: 20,
    }
})