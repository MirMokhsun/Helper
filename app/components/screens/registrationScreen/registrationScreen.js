import React, { Component } from 'react';
import { View, ScrollView, BackHandler, Text, KeyboardAvoidingView, StatusBar } from 'react-native';
import { styles } from './styles';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { CommonInput } from '../../commonComponents/commonInput/commonInput';
import { filterEmail, filterName, filterPhone } from '../../../api/regexFilters/regexFilters';
import { localization } from '../../../common/localization';
import {
    setRegistrationCheckbox,
    setRegistrationPhone,
    setRegistrationCountry,
    setRegistrationEmail,
    setRegistrationFirstName,
    setRegistrationLastName
} from './redux/registrationDataAction';
import CountryPicker from './registrationElements/countryPicker/countryPicker';
import PhoneInput from './registrationElements/phoneInput/phoneinput';
import AgreementAndCheckbox from './registrationElements/agreementAndCheckbox/agreementAndCheckbox';
import ButtonSubmit from '../../commonComponents/buttonSubmit/buttonSubmit';
import AuthorizationButton from './registrationElements/authorizationButton/authorizationButton';
import Logo from '../../commonComponents/svg/logo';
import Background from '../../commonComponents/svg/Background';
import { crmRegistration } from '../../../api/registration/registrationSingleton';
import { showRegistrationSpiner } from '../../commonComponents/buttonSubmit/redux/spinersAction';
import Utils from '../../../api/utils/utils';
import LoginFacebook from './registrationElements/facebookAuth';
import { NavigationEvents } from 'react-navigation';

const { isIOS } = Utils;

export class RegistrationScreen extends Component {
    static navigationOptions = () => {
        return {
            gesturesEnabled: false,
            header: null,
        }
    };

    constructor(props) {
        super(props);
        this.state = {
            isRegistrationButtomDisabled: true,
        }
    }

    componentDidMount = () => {
        BackHandler.addEventListener('hardwareBackRegistrationScreen', this.hardwareBackRegistrationScreen);
        this.clearRegistrationFields();
    }

    UNSAFE_componentWillReceiveProps(nextProps) {
        const { email, firstName, lastName, country, phone, agreementCheckbox } = nextProps.registrationData;
        let isRegistrationButtomDisabled = true;
        if (email.isValid && firstName.isValid && lastName.isValid && country.isValid && phone.isValid && agreementCheckbox.isValid && nextProps.isConnected) {
            isRegistrationButtomDisabled = false;
        }
        this.setState({ isRegistrationButtomDisabled });
    }

    hardwareBackRegistrationScreen = () => {
        const { navigation } = this.props;
        if (navigation.isFocused()) {
            BackHandler.exitApp();
        }
    }

    clearRegistrationFields = () => {
        const { setRegistrationEmail,
            setRegistrationFirstName,
            setRegistrationLastName,
            setRegistrationPhone,
            setRegistrationCheckbox,
            registrationData
        } = this.props;
        setRegistrationFirstName({ isValid: false, value: null });
        setRegistrationEmail({ isValid: false, value: null });
        setRegistrationLastName({ isValid: false, value: null });
        setRegistrationPhone({ isValid: false, value: null });
        if (registrationData.agreementCheckbox.isValid) {
            setRegistrationCheckbox();
        }
    }
    
    getLanguageForCRM = (appsFlyerData, language )=>{
        let lang = ''

        if(appsFlyerData.campaign){
           lang = appsFlyerData.campaign.split("-").pop().toUpperCase()
        }
        else{
            lang = language.toUpperCase()
        }
        return lang
    };

    sendRegistrationData = () => {
        const { registrationData: { email, firstName, lastName, country, phone, agreementCheckbox, phoneCode }, showRegistrationSpiner, appsFlyerData, language, navigation } = this.props;
        if (firstName.value && lastName.value && email.value && agreementCheckbox.isValid && country.countryKey && phone.value && phoneCode && language) {
            crmRegistration.registrationLead(firstName.value, lastName.value, email.value, agreementCheckbox.isValid, country.countryKey, phoneCode, phone.value, this.getLanguageForCRM(appsFlyerData, language), appsFlyerData);
        }
        showRegistrationSpiner();
        navigation.navigate("TabAppNavigator")
    }

    render() {
        const { setRegistrationCheckbox, setRegistrationFirstName, setRegistrationLastName, setRegistrationEmail, setRegistrationPhone,
            registrationData: { phoneCode, email, firstName, lastName, country, phone, agreementCheckbox }, registrationSpiner, language } = this.props;
        const { isRegistrationButtomDisabled } = this.state;
        const local = localization[language];
        return (
            <KeyboardAvoidingView style={styles.wrapperAbsolute} behavior={isIOS ? "padding" : null} enabled >
                <StatusBar barStyle="light-content" backgroundColor="#000756" />
                <ScrollView style={styles.scrollview} keyboardShouldPersistTaps="handled" contentContainerStyle={{ alignItems: 'center', }}>
                    <View style={styles.backgroundBlue}>
                        <Background />
                    </View>
                    <View style={styles.logoWrapper}>
                        <Logo />
                    </View>
                    <View style={styles.container}>
                        <Text style={styles.registrationTitle}>{local.createAccount}</Text>
                        <CommonInput
                            placeholder={local.placeHolder.name}
                            errorText={local.errors.errorFirstName}
                            filter={filterName}
                            action={setRegistrationFirstName}
                            isValid={firstName.isValid}
                            testID="InputFirstNameRegistrationID"
                            textErrorID="FirstNameErrorRegostrationID"
                            currentLanguageName={language}
                        />
                        <CommonInput
                            placeholder={local.placeHolder.lastName}
                            errorText={local.errors.errorLastName}
                            filter={filterName}
                            action={setRegistrationLastName}
                            isValid={lastName.isValid}
                            testID="InputLastNameRegistrationID"
                            textErrorID="LastNameErrorRegostrationID"
                            currentLanguageName={language}
                        />
                        <CommonInput
                            keyboardType="email-address"
                            placeholder={local.placeHolder.email}
                            errorText={local.errors.errorEmail}
                            filter={filterEmail}
                            action={setRegistrationEmail}
                            isValid={email.isValid}
                            testID="InputEmailRegistrationID"
                            textErrorID="EmailErrorRegostrationID"
                            currentLanguageName={language}
                        />
                        <CountryPicker value={country.value} isValid={country.isValid} countryKey={country.countryKey} language={language} />
                        <PhoneInput
                            phcode={phoneCode}
                            maxWidth={230}
                            placeholder={local.placeHolder.phoneNumber}
                            errorText={local.errorMessage.phoneNumber}
                            filter={filterPhone}
                            action={setRegistrationPhone}
                            isValid={phone.isValid}
                            currentLanguageName={language}
                        />
                        <AgreementAndCheckbox
                            title={local["user agreement2"]}
                            isChecked={agreementCheckbox.isValid}
                            setCheckedBox={setRegistrationCheckbox}
                            textWithClick={local["user agreement"]}
                            text={local["I have read and agree with"]}
                        />
                        <ButtonSubmit
                            text={local.Registration}
                            testID="ButtonSubmitRegistrationID"
                            isDisabled={isRegistrationButtomDisabled}
                            isShowSpiner={registrationSpiner}
                            onPress={this.sendRegistrationData}
                        />
                    </View>
                    <LoginFacebook/>
                    <AuthorizationButton
                        text={local["Already have an account?"]}
                        textLogin={local.login}
                    />
                </ScrollView>
            </KeyboardAvoidingView>
        )
    }
}

const mapStateToProps = state => ({
    appsFlyerData: state.appsFlyerData.appsFlyerData,
    language: state.currentLanguage.language,
    registrationData: state.registrationData,
    registrationSpiner: state.spiners.registrationSpiner,
    isConnected: state.connectionStatus.isConnected,
});

const mapDispatchToProps = dispatch => bindActionCreators({
    setRegistrationLastName,
    setRegistrationFirstName,
    setRegistrationEmail,
    setRegistrationCountry,
    setRegistrationPhone,
    setRegistrationCheckbox,
    showRegistrationSpiner
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(RegistrationScreen);