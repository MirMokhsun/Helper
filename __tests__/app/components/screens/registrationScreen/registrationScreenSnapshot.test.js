import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { RegistrationScreen } from '../../../../../app/components/screens/registrationScreen/registrationScreen';

const mock = { 
    phoneCode:true,
    email:true,
    firstName:true,
    lastName:true,
    country:true,
    phone:true,
    agreementCheckbox:true 
}

it('create snapshot RegistrationScreen', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<RegistrationScreen
        setRegistrationLastName={() => { }}
        setRegistrationFirstName={() => { }}
        setRegistrationEmail={() => { }}
        setRegistrationCountry={() => { }}
        setRegistrationPhone={() => { }}
        language={"ru"}
        registrationData = {mock}
        firstName={{ isValid:false }}
        isValid ={false}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});