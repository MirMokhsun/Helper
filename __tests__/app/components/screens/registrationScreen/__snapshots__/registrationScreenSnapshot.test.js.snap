// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot RegistrationScreen 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "backgroundColor": "#E5E5E5",
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <StatusBar
    animated={false}
    backgroundColor="#000756"
    barStyle="light-content"
    showHideTransition="fade"
  />
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "alignItems": "center",
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "top": -50,
        }
      }
    >
      <Background />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 35,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 60,
          "width": "100%",
        }
      }
    >
      <Logo />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "flexDirection": "column",
          "justifyContent": "space-around",
          "minHeight": 933.8,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000756",
            "fontFamily": "Roboto",
            "fontSize": 22,
            "fontWeight": "bold",
            "lineHeight": 32,
            "marginVertical": 25,
            "textAlign": "center",
          }
        }
      >
        Cоздать аккаунт
      </Text>
      <CommonInput
        action={[Function]}
        currentLanguageName="ru"
        errorText="Ошибка ввода имени"
        filter={/\\(\\?=\\(\\^\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"#'!\\?\\$%\\^&\\*_=\\+-\\]\\)\\)\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:@"#!\\?\\$%\\^&\\*_=\\+\\]\\{0,199\\}\\(\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"#!\\?\\$%'\\^&\\*_=\\+-\\]\\$\\)/}
        placeholder="Имя"
        testID="InputFirstNameRegistrationID"
        textErrorID="FirstNameErrorRegostrationID"
      />
      <CommonInput
        action={[Function]}
        currentLanguageName="ru"
        errorText="Ошибка ввода фамилии"
        filter={/\\(\\?=\\(\\^\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"#'!\\?\\$%\\^&\\*_=\\+-\\]\\)\\)\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:@"#!\\?\\$%\\^&\\*_=\\+\\]\\{0,199\\}\\(\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"#!\\?\\$%'\\^&\\*_=\\+-\\]\\$\\)/}
        placeholder="Фамилия"
        testID="InputLastNameRegistrationID"
        textErrorID="LastNameErrorRegostrationID"
      />
      <CommonInput
        action={[Function]}
        currentLanguageName="ru"
        errorText="Ошибка ввода e-mail"
        filter={/\\^\\(\\(\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"\\]\\+\\(\\\\\\.\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"\\]\\+\\)\\*\\)\\|\\("\\.\\+"\\)\\)@\\(\\(\\\\\\[\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\]\\)\\|\\(\\(\\[a-zA-Z\\\\-0-9\\]\\+\\\\\\.\\)\\+\\[a-zA-Z\\]\\{2,\\}\\)\\)\\$/}
        keyboardType="email-address"
        placeholder="Email"
        testID="InputEmailRegistrationID"
        textErrorID="EmailErrorRegostrationID"
      />
      <withNavigation(CountryPicker)
        language="ru"
      />
      <PhoneInput
        action={[Function]}
        currentLanguageName="ru"
        errorText="Неверно указан номер телефона"
        filter={/\\^\\(\\(\\[0-9\\+\\\\-\\(\\)\\]\\)\\{6,24\\}\\)\\$/}
        maxWidth={230}
        phcode={true}
        placeholder="Телефон"
      />
      <withNavigation(AgreementAndCheckbox)
        text="Заполняя форму регистрации, я принимаю"
        textWithClick="условия обработки персональных данных."
        title="Пользовательское соглашение"
      />
      <ButtonSubmit
        isDisabled={true}
        onPress={[Function]}
        testID="ButtonSubmitRegistrationID"
        text="Регистрация"
      />
    </View>
    <withNavigation(AuthorizationButton)
      text="Уже есть аккаунт ?"
      textLogin="Войти"
    />
  </ScrollViewMock>
</KeyboardAvoidingView>
`;
